parser/mpFuncCommon.cpp

```diff
@@ -458,6 +458,34 @@ MUP_NAMESPACE_START
     throw ParserError(err);
   }
 
+  string_type localized_weekday(int week_day, const ptr_val_type *a_pArg) {
+    string_type locale = a_pArg[1]->GetString();
+    string_type ret = "";
+    string_type localized_weekdays[8][7] = {
+      {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
+      {"Søndag", "Mandag", "Tirsdag", "Onsdag", "Torsdag", "Fredag", "Lørdag"},
+      {"Domingo", "Segunda-Feira", "Terça-feira", "Quarta-feira", "Quinta-feira", "Sexta-feira", "Sábado"},
+      {"Domingo", "Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sabado"},
+      {"Dimanche", "Lundi", "Mardi", "Mercredi", "Jeudi", "Vendredi", "Samedi"},
+      {"Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag"},
+      {"星期天", "星期一", "星期二", "星期三", "星期四", "星期五", "星期六"},
+      {"วันอาทิตย์", "วันจันทร์", "วันอังคาร", "วันพุธ", "วันพฤหัสบดี", "วันศุกร์", "วันเสาร์"}
+    };
+    string_type locales[8] = {"en", "nb", "pt-BR", "es-ES", "fr-FR", "de-DE", "zh-CN", "th-TH"};
+
+    for (int i = 0; i < 8; i++) {
+      if(locale == locales[i]) {
+        ret = localized_weekdays[i][week_day];
+      }
+    }

```

Comment:
you could use a map here to achieve O(1). `int index = locales[locale]`

Reply:
i know, but map lookup is only amortized O(1). with n=8, you can see the difference is infinitesimal (in this print array method wins by 4ms but it's basically the same, i ran it a lot of times to check
![image](https://github.com/user-attachments/assets/e5ddf68e-520c-4443-ab08-ec979f536454)

therefore, i don't think the inclusion of the lib is warrented.

scripts used:
# string_method
```cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
  string localized_weekdays[8][7] = {
      {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
      {"Søndag", "Mandag", "Tirsdag", "Onsdag", "Torsdag", "Fredag", "Lørdag"},
      {"Domingo", "Segunda-Feira", "Terça-feira", "Quarta-feira", "Quinta-feira", "Sexta-feira", "Sábado"},
      {"Domingo", "Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sabado"},
      {"Dimanche", "Lundi", "Mardi", "Mercredi", "Jeudi", "Vendredi", "Samedi"},
      {"Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag"},
      {"星期天", "星期一", "星期二", "星期三", "星期四", "星期五", "星期六"},
      {"วันอาทิตย์", "วันจันทร์", "วันอังคาร", "วันพุธ", "วันพฤหัสบดี", "วันศุกร์", "วันเสาร์"}
    };
  string locales[8] = {"en", "nb", "pt-BR", "es-ES", "fr-FR", "de-DE", "zh-CN", "th-TH"};

  string locale, ret;
  int week_day;
  while(cin >> locale >> week_day){
    for (int i = 0; i < 8; i++) {
      if(locale == locales[i]) {
        ret = localized_weekdays[i][week_day];
      }
    }
    cout << ret << endl;
  }
}
```

# map method

```cpp
#include <iostream>
#include <map>
#include <string>
#include <vector>

using namespace std;

int main() {
  map<string, vector<string>> localized_weekdays = {
      {"en", {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"}},
      {"nb", {"Søndag", "Mandag", "Tirsdag", "Onsdag", "Torsdag", "Fredag", "Lørdag"}},
      {"pt-BR", {"Domingo", "Segunda-Feira", "Terça-feira", "Quarta-feira", "Quinta-feira", "Sexta-feira", "Sábado"}},
      {"es-ES", {"Domingo", "Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sabado"}},
      {"fr-FR", {"Dimanche", "Lundi", "Mardi", "Mercredi", "Jeudi", "Vendredi", "Samedi"}},
      {"de-DE", {"Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag"}},
      {"zh-CN", {"星期天", "星期一", "星期二", "星期三", "星期四", "星期五", "星期六"}},
      {"th-TH", {"วันอาทิตย์", "วันจันทร์", "วันอังคาร", "วันพุธ", "วันพฤหัสบดี", "วันศุกร์", "วันเสาร์"}}
  };

  string locale, ret;
  int week_day;

  while(cin >> locale >> week_day){
    ret = localized_weekdays[locale][week_day];
    cout << ret << endl;
  }
}
```

# timer.sh

```bash
#!/bin/bash

# Run the array method 1 million times
echo "Running the array method 1 million times"
start_time1=$(date +%s%N) # Record start time
./array_method < input.txt > /dev/null
end_time1=$(date +%s%N) # Record end time
total_time1=$(((end_time1 - start_time1)/1000000))
echo "Time taken for array_method: $total_time1 miliseconds"

# Run the map method 1 million times
echo "Running the map method 1 million times"
start_time2=$(date +%s%N) # Record start time
./map_method < input.txt > /dev/null
end_time2=$(date +%s%N) # Record end time
total_time2=$(((end_time2 - start_time2)/1000000))
echo "Time taken for map method: $total_time2 miliseconds"
```

Reply:
Nice bench! Lets keep it simple than!